src/platform/linux/input.cpp

```diff
@@ -1756,6 +1812,10 @@ namespace platf {
     libevdev_enable_event_type(dev.get(), EV_ABS);
     libevdev_enable_event_code(dev.get(), EV_ABS, ABS_X, &absx);
     libevdev_enable_event_code(dev.get(), EV_ABS, ABS_Y, &absy);
+    libevdev_enable_event_code(dev.get(), EV_ABS, ABS_MT_SLOT, &mtslot);
+    libevdev_enable_event_code(dev.get(), EV_ABS, ABS_MT_TRACKING_ID, &trkid);
+    libevdev_enable_event_code(dev.get(), EV_ABS, ABS_MT_POSITION_X, &absx);
+    libevdev_enable_event_code(dev.get(), EV_ABS, ABS_MT_POSITION_Y, &absy);
```

Comment:
You should use a separate uinput device for pen/touch input, create it on demand (on the first pen/touch event), and store it in the `client_input_t`. You can use the Windows implementation as reference.